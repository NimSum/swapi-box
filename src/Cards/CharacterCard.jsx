import React from 'react';

export const CharacterCard = cards => 
  cards.map(character => (
    <article className="character-card">
      <h3>{ character.name }</h3>
      <h4>{ character.species }</h4>
      <div>
        <h4>{ character.homeworld }</h4>
        <p>{ character.population }</p>
      </div>
    </article>
  ))
