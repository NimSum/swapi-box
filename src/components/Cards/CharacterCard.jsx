import React from 'react';
import PropTypes from 'prop-types';
import { FavoriteCardBtn } from '../Buttons/Buttons';

export const CharacterCard = ({ card, updateFavoriteCount }) => 
  (
    <article key={ card.id } className="character-card">
      < FavoriteCardBtn 
       card={ card }
       updateFavoriteCount={updateFavoriteCount} />
      <h3>{ card.name }</h3>
      <h4>{ card.speciesName }</h4>
      <div className="home-planet">
        <h4>{ card.worldName }</h4>
        <p>{ card.worldPopulation }</p>
      </div>
    </article>
  )

CharacterCard.propTypes = {
  cards: PropTypes.array
}