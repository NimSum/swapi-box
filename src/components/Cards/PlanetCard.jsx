import React from 'react';
import { FavoriteCardBtn } from '../Buttons/Buttons';
import PropTypes from 'prop-types';

export const PlanetCard = ({ card, updateFavoriteCount }) => {
  const randomBgNum = Math.floor(Math.random() * 7) + 1;
  return (
    <article key={ card.id } className="planet-card">
      < FavoriteCardBtn  
        updateFavoriteCount={updateFavoriteCount}
        card={ card }/>
      <h3>{ card.name }</h3>
      <div className="terrain-container">
        <h4>Terrain:</h4>
        { card.terrain.split(',').map(terrain => (<p>{ terrain }</p>)) }
      </div>
      <h4>Climate:</h4>
      <p>{ card.climate }</p>  
      <h4>Population</h4>    
      <p>{ card.population }</p>
      <div className="residents">
        <h4>Residents:</h4>
        { card.residentNames.length 
          ? card.residentNames.map(person => (
            <p key={person.id}>{ person.name }</p>))
          : (<p>None</p>) }
      </div>
      <img className="bg-img" alt='card background' src={require(`../../images/card-backgrounds/planet-${randomBgNum}.png`)} />
    </article>
  )
}
PlanetCard.propTypes = {
  card: PropTypes.object,
  updateFavoriteCount: PropTypes.func
}
