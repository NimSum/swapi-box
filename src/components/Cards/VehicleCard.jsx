import React from 'react';
import { FavoriteCardBtn } from '../Buttons/Buttons';

export const VehicleCard = ({ cards, updateFavoriteCount }) => 
  cards.map(vehicle => (
    <article className="vehicle-card">
      < FavoriteCardBtn 
        card={ vehicle }
        updateFavoriteCount={updateFavoriteCount}
        type="vehicle" />
      <h3>{ vehicle.name }</h3>
      <div className="vehicle-specs">
        <p>{ vehicle.model }</p>
        <p>{ vehicle.vehicle_class }</p>
        <p>{ parseInt(vehicle.passengers) + parseInt(vehicle.crew) }</p>
      </div>
    </article>
  ))
