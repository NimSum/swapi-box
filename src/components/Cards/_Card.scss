section.card-container {
  article {
    border-radius: 20px;
    width: 100%;
    min-width: 300px;
    margin: 0 auto;
    box-shadow: $card-shadow;
    position: relative;
    background-color: $card-bg-opacity;

    h3, h4, div, button, p {
      position: relative;
      z-index: 1;
    }

    h3, h4 {
      color: $font-card-title-color;
      font-family: $title-font;
      margin: 10px;
    }

    h3 {
      border-width: 2px 0 2px 0;
      border-style: solid;
      border-color: $card-title-border-color;
      padding: 10px 0;
      font-size: 1.7rem;
    }

    p {
      font-family: $content-font;
      margin: 2px auto;
      width: 50%;
    
    }

    img.bg-img {
      position: absolute;
      bottom: 0;
      z-index: 0;
      opacity: .1;
    }
  }


  article.character-card {
    max-height: 350px;

    div.home-planet {
      border-top: 2px solid $card-title-border-color;
      width: 30%;
      white-space: nowrap;
      margin: auto;
    }

    img.bg-img {
      width: 200px;
      left: 0;
      opacity: .2;
    }
  }

  article.vehicle-card {
    min-height: 300px;
    max-height: 350px;
    img.bg-img {
      width: 100%;
      left: 0;
      opacity: .2;
    }
  }

  article.planet-card {
    min-height: 450px;
    max-height: 600px;
    overflow: scroll;

    img.bg-img {
      width: 90%;
      top: 15%;
      left: 0;
      margin: 0 5%;
      opacity: .3;
    }
  }
}