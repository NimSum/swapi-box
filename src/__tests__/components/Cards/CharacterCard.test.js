import { mount } from 'enzyme';
import React from 'react';
import { CharacterCard } from '../../../components/Cards/CharacterCard';

describe('Character card', () => {
  const mockCard = { 
    id: 123, 
    name: 'Nimsumm', 
    speciesName: 'Human', 
    worldName: 'Earth', 
    worldPopulation: '7500000000'
  }
  const mockupdateFavoriteCount = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      < CharacterCard 
        card={ mockCard }
        updateFavoriteCount={ mockupdateFavoriteCount }/>
    )
  })

  it('Should match snapshot with card passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

})