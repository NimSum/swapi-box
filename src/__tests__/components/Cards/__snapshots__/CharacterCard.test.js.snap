// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character card Should match snapshot with card passed in 1`] = `
<CharacterCard
  card={
    Object {
      "id": 123,
      "name": "Nimsumm",
      "speciesName": "Human",
      "worldName": "Earth",
      "worldPopulation": "7500000000",
    }
  }
  updateFavoriteCount={[MockFunction]}
>
  <article
    className="character-card"
    key="123"
  >
    <FavoriteCardBtn
      card={
        Object {
          "id": 123,
          "name": "Nimsumm",
          "speciesName": "Human",
          "worldName": "Earth",
          "worldPopulation": "7500000000",
        }
      }
      updateFavoriteCount={[MockFunction]}
    >
      <button
        className="favorite-card-btn"
        name="favorite-card"
        onClick={[Function]}
        title="Add to Favorites"
        type="button"
      >
        <img
          alt="card background"
          src="empire.png"
        />
        <img
          alt="card background"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "inactive-star.svg",
            }
          }
        />
        <img
          alt="card background"
          src="rebel-alliance.png"
        />
      </button>
    </FavoriteCardBtn>
    <h3>
      Nimsumm
    </h3>
    <h4>
      Human
    </h4>
    <div
      className="home-planet"
    >
      <h4>
        Home Planet:
      </h4>
      <p
        className="world-name"
      >
        Earth
      </p>
      <p>
        7500000000
      </p>
    </div>
    <img
      alt="card background"
      className="bg-img"
      src="character-bg.png"
    />
  </article>
</CharacterCard>
`;
